### PR TITLE
Properly parse IPv6 addresses in V_BACKUP_SERVER_NODE

### PR DESCRIPTION
**Pull request by jonathanl-bq**
This PR fixes a bug where IPv6 addresses weren't being parsed correctly when specified in the V_BACKUP_SERVER_NODE environment variable. I used parser combinators here to make parsing individual components composable and very easy to unit test. As a result, I also added several unit tests and included Danny's integration test.

### CODE REVIEW

File: packages/vertica-nodejs/lib/backup-nodes-parsers.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  var result = ""  
+    
+  for (let c of input) {
+    if (c === ':') {
+      break
+    }
+    result += c
+  }
+    
+  if (result.includes('.')) {
```
Comment (williammatherjones) [inline packages/vertica-nodejs/lib/backup-nodes-parsers.js:81]: Does this need to distinguish 10.1.1.1 from vertica.com?   They both contain '.'

File: packages/vertica-nodejs/lib/backup-nodes-parsers.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  var hostname = ""
+    
+  for (let c of input) {
+    if (c === ':') {
+      break
+    }
+    hostname += c
+  }
+   
+  return [input.substring(hostname.length), hostname]
```
Comment (williammatherjones) [inline packages/vertica-nodejs/lib/backup-nodes-parsers.js:130]: Does this distinguish between hostname and IP addresses?

**Review by williammatherjones — APPROVED** [review]

File: packages/vertica-nodejs/lib/backup-nodes-parsers.js (hunk 1 above)
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/backup-nodes-parsers.js:81]: Resolving this as we discussed it in stand-up. The IPv4 case will still catch hostnames with . in them and parse them correctly, although we may still want to properly parse IPv4 addresses, as the function name suggests. Lower priority for now though.

File: packages/vertica-nodejs/lib/backup-nodes-parsers.js (hunk 2 above)
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/backup-nodes-parsers.js:130]: As mentioned in the other comment, we do not really distinguish between hostnames and IP addresses properly, but the current logic happens to work for the cases we care about. More details in the other comment.

**Pull request merged**